### PR TITLE
SSH Agent: Allow toggling integration on and off without restart

### LINKED ISSUE #1196: SSH Agent: Global enable/disable requires a restart
The new SSH agent feature can be enabled/disabled from settings but the current implementation requires restarting KeePassXC for the change to take effect.

## Expected Behavior
Requires discussion.

## Current Behavior
Requires restart.

## Context
@droidmonkey noted in #1098 that all settings should work on-the-fly and not require a restart.

There are a few key issues that need to be addressed:
- When disabling SSH Agent, should remove-key-on-lock be triggered for all open databases?
- When enabling SSH Agent, should add-key-on-unlock be triggered for all open databases?
- EditEntryWidget needs to be refreshed somehow to add/remove the SSH Agent widget

I guess the question here is, what would the user expect?

### PR DESCRIPTION
**Pull request by hifi**
[TIP]:  # ( Provide a general summary of your changes in the title above ^^ )
After refactors this was rather trivial. When integration is disabled all remove-on-lock keys are removed from the agent as a safety measure.

This PR includes a hefty cleanup of the SSHAgent singleton and configuration management.

## Type of change
[NOTE]: # ( Please remove all lines which don't apply. )
- ✅ Bug fix (non-breaking change which fixes an issue)
- ✅ Refactor (significant modification to existing code)
- ✅ New feature (non-breaking change which adds functionality)

## Description and Context
[NOTE]: # ( Describe your changes in detail, why is this change required? )
[NOTE]: # ( Describe the context of your change. Explain large code modifications. )
[NOTE]: # ( If it fixes an open issue, please add "Fixes #XXX" as necessary )
Fixes #1196 and an old regression from commit 9477437 which could only be seen as the history tab being visible even if there is no history for an entry.

## Testing strategy
[NOTE]: # ( Please describe in detail how you tested your changes. )
[TIP]:  # ( We expect new code to be covered by unit tests and documented with doc blocks! )
Both changes tested on Linux by hand.

## Checklist:
[NOTE]: # ( Please go over all the following points. )
[NOTE]: # ( Again, remove any lines which don't apply. )
[NOTE]: # ( Pull Requests that don't fulfill all [REQUIRED] requisites are likely )
[NOTE]: # ( to be sent back to you for correction or will be rejected.  )
- ✅ I have read the **CONTRIBUTING** document. **[REQUIRED]**
- ✅ My code follows the code style of this project. **[REQUIRED]**
- ✅ All new and existing tests passed. **[REQUIRED]**
- ✅ I have compiled and verified my code with `-DWITH_ASAN=ON`. **[REQUIRED]**

### CODE REVIEW

File: src/sshagent/SSHAgent.cpp [hunk 1]
```diff
@@ -29,46 +29,52 @@
 #include <windows.h>
 #endif
 
-SSHAgent* SSHAgent::m_instance;
+Q_GLOBAL_STATIC(SSHAgent, g_sshAgent);
```
Comment (phoerious) [inline src/sshagent/SSHAgent.cpp]: This should be named `s_sshAgent` or `s_instance` or something along the lines.

File: src/gui/EditWidget.cpp [hunk 2]
```diff
@@ -70,7 +70,16 @@ void EditWidget::addPage(const QString& labelText, const QIcon& icon, QWidget* w
...
+    int index = -1;
+
+    for (int i = 0; i < m_ui->stackedWidget->count(); i++) {
+        auto* scrollArea = qobject_cast<QScrollArea*>(m_ui->stackedWidget->widget(i));
+        if (scrollArea != nullptr && scrollArea->widget() == widget) {
+            index = i;
+            break;
+        }
+    }
+
```
[the remark this answers is not in the document]
Comment (hifi) [reply src/gui/EditWidget.cpp]: The real widget is wrapped inside the `QScrollArea` and we need to know which widget it was to properly hide a tab. See the function above.

[the remark this answers is not in the document]
Comment (hifi) [reply src/gui/EditWidget.cpp]: For reference we discussed this on IRC and the current loop is good enough for now.

**Comment by hifi** [comment]
This should be ready for the next review round. :crossed_fingers:

File: src/gui/EditWidget.cpp [hunk 3]
```diff
@@ -70,7 +70,16 @@ void EditWidget::addPage(const QString& labelText, const QIcon& icon, QWidget* w
 
 void EditWidget::setPageHidden(QWidget* widget, bool hidden)
 {
-    int index = m_ui->stackedWidget->indexOf(widget);
+    int index = -1;
+
+    for (int i = 0; i < m_ui->stackedWidget->count(); i++) {
+        auto* scrollArea = qobject_cast<QScrollArea*>(m_ui->stackedWidget->widget(i));
+        if (scrollArea != nullptr && scrollArea->widget() == widget) {
```
Comment (droidmonkey) [inline src/gui/EditWidget.cpp]: `if (scrollArea && scrollArea->widget() == widget) {`

File: src/sshagent/SSHAgent.cpp [hunk 4]
```diff
@@ -29,46 +29,68 @@
...
-    m_instance = new SSHAgent(parent);
+    QString socketPath;
+
+#ifndef Q_OS_WIN
+    socketPath = authSockOverride();
+    if (socketPath.isEmpty() || !allowOverride) {
+        socketPath = QProcessEnvironment::systemEnvironment().value("SSH_AUTH_SOCK");
+    }
+#else
+    socketPath = "\\\\.\\pipe\\openssh-ssh-agent";
```
Comment (droidmonkey) [inline src/sshagent/SSHAgent.cpp]: This is hard to parse, can you include a comment that describes what is going on here

Comment (hifi) [reply src/sshagent/SSHAgent.cpp]: Split it up to make it more apparent what is going on. Added a comment as well.
